main.go

```diff
@@ -169,6 +170,27 @@ func _main(ctx context.Context) error {
 	var auxDB *dcrpg.ChainDBRPC
 	var newPGIndexes, updateAllAddresses, updateAllVotes bool
 	if usePG {
+		// proposalsUpdateChan is used to manage politeia update notifications trigger
+		// between the notifier and the handler method.
+		proposalsUpdateChan := make(chan struct{})
```

Comment:
That's a very good point.  Could then simplify `(pgb *ChainDB) proposalsUpdateHandler(proposalsUpdateChan chan struct{}) {` to `(pgb *ChainDB) proposalsUpdateHandler() {` and get the channel inside the goroutine.

Also, the signature `UpdateSignal() (chan struct{})` could be `UpdateSignal() (<-chan struct{})` to ensure the user of the channel can only receive, not send.

main would not need to make the `proposalsUpdateChan` either if `NewParser`did it.  This change is probably the annoying part of this suggested change since piparser doesn't even know about a channel presently (it just has the `updatesTrigger` function).  Are there cases where a channel would not be used in a reasonable updateTrigger functoin?

Reply:
Note that changing the signature of exported methods should be done with care to keep the Go API stable.  Now is the time to make such changes though.  But an approach to avoiding this, when the feature is optional as it is in this case, is to keep the constructor the same as before and add a `UseProposalHistoryFetcher` method.

Reply:
I had the channel out of the piparser to ensure that the users of the tool had more control over the update process and ensure that the channel never got blocked in case one doesn't want to fetch the updates. I could still try to much of the implementation into the piparser as requested.

Reply:
Let's just follow up on it later.

Reply:
The fix just as suggested is almost done.